kiro-synth-core/src/effects/delay.rs

```diff
@@ -0,0 +1,77 @@
+use crate::float::Float;
+use std::collections::VecDeque;
```

Comment:
We need to use a collection that doesn't use the heap to keep compatibility with microcontrollers.